### PR TITLE
[Feature] Support 'folder' mode for computing average metric for video SR models

### PR DESCRIPTION
**Pull request by ckkelvinchan**
When computing quantitative metrics (e.g. PSNR) for video SR, we usually compute the PSNR for each folder first, and then compute the average of the averages". In the existing `SRVid4Dataset`, only the average of all frames is supported. 

This PR adds support of computing the average of averages by adding an argument `metric_average_mode`.

### CODE REVIEW

**Comment by ckkelvinchan** [comment]
Will add more unittest.

File: mmedit/datasets/sr_vid4_dataset.py [hunk 1]
```diff
@@ -76,3 +93,49 @@ def load_annotations(self):
...
+        """Evaluate with different metrics.
+
+        Args:
+            results (list[tuple]): The output of forward_test() of the model.
+
+        Return:
+            dict: Evaluation results dict.
+        """
+        if not isinstance(results, list):
+            raise TypeError(f'results must be a list, but got {type(results)}')
```
Comment (Yshuo-Li) [inline mmedit/datasets/sr_vid4_dataset.py:107]: The error needs to be triggered in the unit test.

**Review by Yshuo-Li — COMMENTED** [review]
Need update master and rebase

**Review by Yshuo-Li — APPROVED** [review]

File: mmedit/datasets/sr_vid4_dataset.py [hunk 2]
```diff
@@ -33,6 +37,10 @@ class SRVid4Dataset(BaseSRDataset):
         scale (int): Upsampling scale ratio.
         filename_tmpl (str): Template for each filename. Note that the
             template excludes the file extension. Default: '{:08d}'.
+        metric_average_mode (str): The way to compite the average metric.
```
Comment (innerlee) [inline mmedit/datasets/sr_vid4_dataset.py:40]: compite

File: mmedit/datasets/sr_vid4_dataset.py [hunk 3]
```diff
@@ -45,6 +53,7 @@ def __init__(self,
                  pipeline,
                  scale,
                  filename_tmpl='{:08d}',
+                 metric_average_mode='folder',
```
Comment (innerlee) [inline mmedit/datasets/sr_vid4_dataset.py:56]: What does the frames in a "folder" represent? A video/clip? If so, then the name should be "video" or "clip".
The metric should be independent of its data representation. Say if we implement an evaluation with raw video input, then the "folder" will be strange by then.

File: mmedit/datasets/sr_vid4_dataset.py [hunk 4]
```diff
@@ -64,16 +64,15 @@ def __init__(self,
...
+        if metric_average_mode not in ['clip', 'all']:
+            raise ValueError('metric_average_mode can only be "clip" or '
                              f'"all", but got {metric_average_mode}.')
 
         self.metric_average_mode = metric_average_mode
         self.data_infos = self.load_annotations()
 
     def load_annotations(self):
         """Load annoations for Vid4 dataset.
-
```
Comment (innerlee) [inline mmedit/datasets/sr_vid4_dataset.py:62]: Do not modify this line

File: mmedit/datasets/sr_vid4_dataset.py [hunk 5]
```diff
@@ -96,10 +95,8 @@ def load_annotations(self):
 
     def evaluate(self, results, logger=None):
         """Evaluate with different metrics.
-
```
Comment (innerlee) [inline mmedit/datasets/sr_vid4_dataset.py]: here

File: mmedit/datasets/sr_vid4_dataset.py [hunk 6]
```diff
@@ -96,10 +95,8 @@ def load_annotations(self):
...
         Args:
             results (list[tuple]): The output of forward_test() of the model.
-
```
Comment (innerlee) [inline mmedit/datasets/sr_vid4_dataset.py]: here

**Pull request merged**